logging/logging.go

```diff
@@ -630,6 +632,9 @@ func jsonValueToStructValue(v interface{}) *structpb.Value {
 // and will block, it is intended primarily for debugging or critical errors.
 // Prefer Log for most uses.
 func (l *Logger) LogSync(ctx context.Context, e Entry) error {
+	if e.Severity < l.minimalLoggedSeverity {
```

Comment:
Not sure if this is valid concern, but since `Default` the lowest severity, we should also introduce `None` to be the lowest severity, thus anyone could also log with `Default` severity which correct me if I wrong a valid cloud Logging severity.

Reply:
I got you. The `Default` severity level is a very strange thing. [Here](https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry#logseverity), the `Default` severity is described as "The log entry has no assigned severity level." We cannot recognize the severity level for "default" logs. Apparently, "default" logs will always have to be logged, regardless of the minimum severity level. I will prepare the changes in the code. Thank you for insight.)

Reply:
Done.

Reply:
Thanks @kimmelserj! 
@daniel-sanche, thinking about it twice I wonder if we could add Severity.None (-1) same way as we did in Java. This way we could make a logic to be straightforward - the only addition would be to make sure that this severity cannot be used in log entries

Reply:
> Not sure if this is valid concern, but since Default the lowest severity, we should also introduce All to be the "include-all" severity, thus anyone could also log with Default severity which correct me if I wrong a valid cloud Logging severity.

My comment is out of date now since I'm late, but I believe the original code with `minimalLoggedSeverity` was already the "include-all" option, since it was filtering logs out using `<` and not `<=`

> Apparently, "default" logs will always have to be logged, regardless of the minimum severity level. I will prepare the changes in the code. Thank you for insight.)

Hmm I'm not sure if I agree with this behaviour. I'd probably say we should treat Default as as a filterable level below debugging with severity 0 like [in the spec](https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry#logseverity). But there's definitely an argument to this change too, so I'm fine either way

> @daniel-sanche, thinking about it twice I wonder if we could add Severity.None (-1) same way as we did in Java. This way we could make a logic to be straightforward - the only addition would be to make sure that this severity cannot be used in log entries

Yeah, it may be something worth considering following Java for the next step if people ask for it, but I'd vote we should keep this PR simple for now and merge this without it.

Reply:
> > Not sure if this is valid concern, but since Default the lowest severity, we should also introduce All to be the "include-all" severity, thus anyone could also log with Default severity which correct me if I wrong a valid cloud Logging severity.
> 
> My comment is out of date now since I'm late, but I believe the original code with `minimalLoggedSeverity` was already the "include-all" option, since it was filtering logs out using `<` and not `<=`
> 
> > Apparently, "default" logs will always have to be logged, regardless of the minimum severity level. I will prepare the changes in the code. Thank you for insight.)
> 
> Hmm I'm not sure if I agree with this behaviour. I'd probably say we should treat Default as as a filterable level below debugging with severity 0 like [in the spec](https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry#logseverity). But there's definitely an argument to this change too, so I'm fine either way
> 
> > @daniel-sanche, thinking about it twice I wonder if we could add Severity.None (-1) same way as we did in Java. This way we could make a logic to be straightforward - the only addition would be to make sure that this severity cannot be used in log entries
> 
> Yeah, it may be something worth considering following Java for the next step if people ask for it, but I'd vote we should keep this PR simple for now and merge this without it.

@kimmelserj , seems adding Severity.None is not a big of a fix - can you please let us know if you can make it?